### PR TITLE
remove java_class_loader_limitt from jar_filet

### PR DESCRIPTION
**Pull request by kroening**
I am trying to clean up the mocking mechanisms that made it into the Java frontend.

As a starter, this pushes up the "loading limit" from jar files up to the class loader.

### CODE REVIEW

File: jbmc/src/java_bytecode/java_class_loader.cpp [hunk 1]
```diff
@@ -116,6 +116,16 @@ java_class_loadert::get_parse_tree(
...
+  // do we refuse to load?
+  if(!class_loader_limit.load_class_file(class_name_to_file(class_name)))
+  {
+    debug() << "not loading " << class_name << " because of limit" << eom;
+    java_bytecode_parse_treet parse_tree;
+    parse_tree.parsed_class.name=class_name;
+    parse_trees.push_back(std::move(parse_tree));
+    return parse_trees;
+  }
+
```
Comment (mgudemann) [inline jbmc/src/java_bytecode/java_class_loader.cpp]: This duplicates the code at the end of the function, but ok with me.
nitpick ` = ` in line 124

**Review by mgudemann — APPROVED** [review]

**Review by smowton — APPROVED** [review]

File: jbmc/src/java_bytecode/java_class_loader.cpp (hunk 1 above)
Comment (kroening) [reply jbmc/src/java_bytecode/java_class_loader.cpp]: Yes, understood, did bug me as well.

**Review by peterschrammel — APPROVED** [review]
Needs a rebase, then it should be fine.

**Comment by mgudemann** [comment]
@kroening restarted failing CI, was just a Travis hic-up

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: c4569dde236ad88f7894752089dcaf5d34c7430c).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/80759097

**Pull request merged**